doc/developer/design/20250226_postgres_style_explain.md

```diff
@@ -0,0 +1,338 @@
+
+We will use LIR as the new default `EXPLAIN`/`EXPLAIN AS TEXT`
+output. We will update `mz_lir_mapping` to use the new Postgres-style
+syntax (fixing [a bug with `Let` and `LetRec`
+rendering](https://github.com/MaterializeInc/database-issues/issues/8993)
+in the process.
+
+We will need three pieces of work, which should all land together:
+
+ - We must implement the new output and put in appropriate SLT tests
+   for it.
+ - We must update `mz_lir_mapping` to use the new vocabulary.
+ - We must update the documentation to explain the new output, ideally
+   using this output everywhere `EXPLAIN` is used.
+
+### Concrete Mapping
+
+| LIR node    | `mz_lir_mapping` node                    | New, Postgres-style syntax              |
+| :---------- | :--------------------------------------- | :-------------------------------------- |
+| `Constant`  | `Constant`                               | `Constant`                              |
+| `Get`       | `Get::PassArrangements l0`               | `Index Scan on l0 using ...`            |
+| `Get`       | `Get::Arrangement l0 (val=...)`          | `Index Lookup on l0 using ...`          |
+| `Get`       | `Get::Collection l0`                     | `Read l0`                               |
+| `Mfp`       | `MapFilterProject`                       | `Map/Filter/Project`                    |
+| `FlatMap`   | `FlatMap`                                | `Flat Map`                              |
+| `Join`      | `Join::Differential`                     | `Differential Join`                     |
+| `Join`      | `Join::Delta`                            | `Delta Join`                            |
+| `Reduce`    | `Reduce::Distinct`                       | `Distinct GroupAggregate`               |
+| `Reduce`    | `Reduce::Accumulable`                    | `Accumulable GroupAggregate`            |
+| `Reduce`    | `Reduce::Hierarchical (monotonic)`       | `Monotonic Hierarchical GroupAggregate` |
+| `Reduce`    | `Reduce::Hierarchical (buckets: ...)`    | `Bucketed Hierarchical GroupAggregate`  |
+| `Reduce`    | `Reduce::Basic`                          | `Non-incremental GroupAggregate`        |
+| `Reduce`    | `Reduce::Collation`                      | `Collated GroupAggregate` (details?)    |
+| `TopK`      | `TopK::MonotonicTop1`                    | `Monotonic Top1`                        |
+| `TopK`      | `TopK::MonotonicTopK`                    | `Monotonic TopK`                        |
+| `TopK`      | `TopK::Basic`                            | `Non-monotonic TopK`                    |
+| `Negate`    | `Negate`                                 | `Negate Diffs`                          |
+| `Threshold` | `Threshold`                              | `Threshold Diffs`                       |
+| `Union`     | `Union`                                  | `Union`                                 |
+| `Union`     | `Union (consolidates output)`            | `Consolidating Union`                   |
```

Comment:
I'm wondering whether users need to know about whether a `Union` consolidates. Seems like a somewhat esoteric implementation-detail, so maybe not. Also, I can't really think of a situation where user would need to make some decision based on knowing whether a Union consolidates or not.

Reply:
Happy to skip it! My understanding was that we sometimes cared about this in accounting for memory footprint, but maybe that's an us/`VERBOSE TEXT` thing?

Reply:
Yeah, it's more for us, enough in `VERBOSE TEXT`. (But even I haven't looked at this in a long time. Currently, the heuristic for whether a Union consolidates is very simple: if there is at least 1 negated input, then we consolidate. If we were to make the heuristic more complex then we'd need to more often look at it. There was an abandoned attempt on this here: https://github.com/MaterializeInc/materialize/pull/30360 )